### ISSUE TITLE
Ability to save PSD's

### ISSUE THREAD

**Issue by mexicantexan** [chinedufn/psd#61]
I'm happy to take this up!             
              
              
              Right now the `psd` crate only supports deserializing a PSD from bytes. It does not currently support writing PSD bytes to a file.

I'd be happy to help guide anyone that is interested in contributing support for this.

_Originally posted by @chinedufn in https://github.com/chinedufn/psd/issues/51#issuecomment-1850061709_

**Comment by chinedufn** [comment]
I would recommend starting with:

- come up with a few very basic writing scenarios. For instance, "saving an empty file", "saving a 1x1 file"
- create a `mod write_bytes` or something like that
- add tests for the basic scenarios
- get those tests passing
- submit a PR

We want to be able to round-trip PSDs such that if we write a PSD, then read it, it equals the exact same in memory Rust PSD that we wrote.
So, our test suite should test that.

---

In short, step 1 here is just getting some tests written for simple cases.

Then from there we can continue to test more and more cases.

Please feel free to continue asking questions.
